pkg/tfgen/installation_docs.go

```
@@ -245,3 +250,106 @@ func convertExample(g *Generator, code string, exampleNumber int) (string, error
+			if child, ok := currentChild.(*ast.Heading); ok && child.Level <= removingLevel {
+				removingLevel = 0
+			} else {
+				nextNode := currentChild.NextSibling()
+				currentChild.Parent().RemoveChild(currentChild.Parent(), currentChild)
+				currentChild = nextNode
+				continue
+			}
+		}
+		// In case our level got reset, we check again
+		if removingLevel == 0 {
+			switch child := currentChild.(type) {
+			case *ast.Heading:
+				header := child.Text(source)
+				if t.shouldSkipHeader(string(header)) {
+					removingLevel = child.Level
+					nextNode := child.NextSibling()
+					child.Parent().RemoveChild(child.Parent(), child)
+					currentChild = nextNode
+					continue
+				}
+			}
+		}
+		// Move to next node in base case.
+		currentChild = currentChild.NextSibling()
+	}
+}
+
+type shouldSkipHeaderFunc = func(headerText string) bool
+
+// SkipSectionByHeaderContent takes a document's content byte array and a callback function that contains logic on
+// whether a given header should be skipped.
+// It uses "github.com/yuin/goldmark" parser in conjunction with "github.com/teekennedy/goldmark-markdown" renderer
+// to walk the markdown document tree and apply the SectionSkipper's transformer
+func SkipSectionByHeaderContent(content []byte, shouldSkipHeader shouldSkipHeaderFunc) ([]byte, error) {
+	// Instantiate our transformer
+	sectionSkipper := SectionSkipper{
+		shouldSkipHeader: shouldSkipHeader,
+	}
+	gm := goldmark.New(
+		goldmark.WithExtensions(extension.GFM),
+		goldmark.WithParserOptions(
+			parser.WithASTTransformers(
+				util.Prioritized(recognizeHeaderAfterHTML{}, 2000),
+				util.Prioritized(sectionSkipper, 500),
+			),
+		),
+		goldmark.WithRenderer(markdown.NewRenderer()),
+	)
+	buf := bytes.Buffer{}
+	// Convert parses the source, applies transformers, and renders output to buf
+	err := gm.Convert(content, &buf)
+	if err != nil {
+		log.Fatalf("Encountered Markdown conversion error: %v", err)
+	}
+	return buf.Bytes(), nil
```

Why would we panic here when we can return the error?
```suggestion
	err := gm.Convert(content, &buf)
	return buf.Bytes(), err
```